### PR TITLE
Add support to use PEM string for SSL

### PR DESCRIPTION
**Pull request by stephen37**
**What this PR does / why we need it**:
Make it possible to use CA certificate string (PEM format) for verifying the broker's key,
Client's private key string (PEM format) used for authentication and Client's public key string (PEM format) used for authentication.

Without that, it is only possible to use SSL with path to files for PEM. 
**Which issue(s) this PR fixes**:

Fixes https://github.com/SeldonIO/seldon-core/issues/3867

**Special notes for your reviewer**:
I'm gonna add some documentation later :)

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @stephen37. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **majolo**
You can assign the PR to them by writing `/assign @majolo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: executor/api/util/utils.go [hunk 1]
```diff
@@ -160,10 +163,16 @@ func GetKafkaSecurityProtocol() string {
...
+		ClientCert: GetEnv("KAFKA_SSL_CLIENT_CERT", ""),
+		ClientKey:  GetEnv("KAFKA_SSL_CLIENT_KEY", ""),
+		CACert:     GetEnv("KAFKA_SSL_CA_CERT", ""),
+		// If we use path to files instead of string
 		ClientCertFile: GetEnv("KAFKA_SSL_CLIENT_CERT_FILE", ""),
 		ClientKeyFile:  GetEnv("KAFKA_SSL_CLIENT_KEY_FILE", ""),
 		CACertFile:     GetEnv("KAFKA_SSL_CA_CERT_FILE", ""),
-		ClientKeyPass:  GetEnv("KAFKA_SSL_CLIENT_KEY_PASS", ""),
+		// Optional password
+		ClientKeyPass: GetEnv("KAFKA_SSL_CLIENT_KEY_PASS", ""),
```
Comment (cliveseldon) [inline executor/api/util/utils.go:174]: I assume if password is set to empy string as no env setting its ok later when its set in Kafka options? Will it be ignored?

**Comment by cliveseldon** [comment]
@stephen37 failing some tests.

**Comment by stephen37** [comment]
@cliveseldon Yes tests won't work because the version of Kafka is too old, I can't fix the test until we use v1.8.2. The error is `No such configuration property: \"ssl.ca.pem\` in the test.

File: executor/api/util/utils.go (hunk 1 above)
Comment (stephen37) [reply executor/api/util/utils.go:174]: Yes, it will be ignored 😃  It has been tested in the `worker_test.go` file

**Comment by cliveseldon** [comment]
> @cliveseldon Yes tests won't work because the version of Kafka is too old, I can't fix the test until we use v1.8.2. The error is `No such configuration property: \"ssl.ca.pem\` in the test.

OK so will you add a PR for that so we can get in and rebase?

**Comment by stephen37** [comment]
Sure, here is the PR https://github.com/SeldonIO/seldon-core/pull/3870, I just assumed it would be more complicated than that to upgrade. Also the tests work locally 🙌

**Comment by stephen37** [comment]
Ah I guess I need you yo do something so that we can check if tests pass now @cliveseldon.

**Comment by stephen37** [comment]
Yaaay tests are passing now 🎉 Only the documentation tests are failing but it seems like they are not working at the moment.

**Comment by cliveseldon** [comment]
@stephen37 Looks good will run longer integration test. There are 1 or 2 flaky ones so lets see how they go. And yes docs tests seems to have an issue at present.

**Comment by seldondev** [comment]
@stephen37: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | d179fbedf7e3de7b39def264ad60a820ea7400bd | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3868-integration-sx7ft) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by stephen37** [comment]
The integration test failed but I don't have access to it, how can I check what failed? @cliveseldon

**Comment by cliveseldon** [comment]
@stephen37 it failed during test setup. We are looking to see if we can open up the logs without auth but for now will rerun tests. There are a couple of flaky tests so if its just those we can proceed as notebook tests succeed.

**Comment by seldondev** [comment]
@stephen37: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | d179fbedf7e3de7b39def264ad60a820ea7400bd | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3868-integration-9v5n5) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by cliveseldon** [comment]
As just 2 tests failing in integration but we know these have issues can merge. Thanks @stephen37

**Pull request merged**